### DISCUSSION TITLE
Upgrading from 2.68 to 2.69 breaks

### DISCUSSION THREAD

**Discussion by dragosde** [ghostfolio/ghostfolio#3216] [Q&A]
Hi guys, I wonder if you can help.
It may just be me, running the ghostfolio docker on an ARM but the new (latest) image breaks something. 
My container now throws 

exec /tini: exec format error 

Reverting to 2.68 restores functionality 

I wonder where I should look for a solution

**Comment by dtslvr** [comment]
I have created an issue: https://github.com/ghostfolio/ghostfolio/issues/3217

Any idea what is happening @Sonlis?

Reference: https://github.com/ghostfolio/ghostfolio/pull/3195

**Reply by Sonlis** [reply]
This is an error that is thrown when the image for a wrong architecture is ran. The changes from the PR are not related to this, I am wondering if the build for ARM architecture broke.

**Reply by Sonlis** [reply]
I'll have a look tonight, I have some ARM devices laying around and check what's wrong.

**Reply by Sonlis** [reply]
Found the problem indeed, the link used to get the binary in the dockerfile is for x86. There are other links for ARM and other platform: https://github.com/krallin/tini/tree/0b44d3665869e46ccbac7414241b8256d6234dc4?tab=readme-ov-file#other-platforms. To build it for both x86 and ARM, we would need to add an argument in the dockerfile so the link would change based on the platform.
Right off the bat, the only way to do it would be to have multiple steps in the github action, one for each architecture, which passes the argument to the dockerfile so it downloads tini for the right architecture.
That makes me think is it worth it to add it to the container image instead of adding it to `docker-compose.yml` and `docker run` commands. I am guessing not many architecture need to be supported, but adding tini straight to the container means having a build step for each architecture.

**Reply by Sonlis** [reply]
There seem to be some ARG in the docker image which gives information at build time about the target platform: https://docs.docker.com/reference/dockerfile/#automatic-platform-args-in-the-global-scope. I can check tomorrow if they match the download link of tini for each platform, and if they do then it's just a matter of modifying the download link here: https://github.com/ghostfolio/ghostfolio/blob/a2277dea2c631548e7f8c1f60380eb49ff3518d4/Dockerfile#L62 to include the target platform argument.

**Reply by dtslvr** [reply]
> I can check tomorrow if they match the download link of tini for each platform

Okay.

If that is not feasible, I would suggest to revert the change in the `Dockerfile` and make a proposal for a `docker-compose.yml` only solution.

**Reply by Sonlis** [reply]
I tried with the arguments, but unfortunately the platform `linux/arm/v7` does not have a matching architecture in the tini repository. I will revert the Dockerfile changes and add it to `docker-compose.yml`.
